demisto_sdk/commands/generate_docs/generate_integration_doc.py

```diff
@@ -337,29 +338,42 @@ def generate_command_example(cmd, cmd_example=None):
     return example, errors
 
 
-def get_command_examples(commands_file_path):
+def get_command_examples(commands_file_path, specific_commands):
     """
     get command examples from command file
 
     @param commands_file_path: command file or the content of such file
+    @param specific_commands: commands specified by the user
 
     @return: a list of command examples
     """
-    commands = []  # type: list
+    command_examples = []  # type: list
 
     if commands_file_path is None:
-        return commands
+        return command_examples
 
     if os.path.isfile(commands_file_path):
         with open(commands_file_path, 'r') as examples_file:
-            commands = examples_file.read().splitlines()
+            command_examples = examples_file.read().splitlines()
     else:
         print('failed to open command file')
-        commands = commands_file_path.split('\n')
+        command_examples = commands_file_path.split('\n')
+
+    commands = []
+    if specific_commands:
+        for command_ex in command_examples:
+            is_command_specific = False
+            for specific_command in specific_commands:
+                if command_ex.startswith(f'!{specific_command}'):
+                    is_command_specific = True
```

Comment:
append here the command instead of using the flag - it is much more readable. you can also ditch the for and create a filter that validates the commands in the example are exists in the commands instead.

Reply:
done.